### PR TITLE
chore: create script for updating RELEASE.md before PR

### LINKED ISSUE #148: chore: create script for updating RELEASE.md before PR
✨ Desired outcome: easily update correct RELEASE.md file for a change that a PR will bring

Running the script should be able to
1. ask for scope and type of change the PR will bring in
2. correctly identify the appropriate RELEASE.md file
3. ask for message
4. write message to appropriate section of the file (fix or feat)

The script could later be used for automating update of change logs depending on PR title :)

### PR DESCRIPTION
**Pull request by NeilAn99**
## chore: create script for updating RELEASE.md before PR

Fixes #148 

Description of changes: 
- Added the [prompt-sync](https://www.npmjs.com/package/prompt-sync) and [semver](https://www.npmjs.com/package/semver) packages.
- Created a script `pr-updater.js` located in the `packages` directory that can be run using `npm run release`. This script will prompt the contributor for the package changed, the type of PR, and the description(s) of changes made. Then, the script will find the corresponding RELEASE.md file and update it.

For example, the running the script:
image
Will produce the following RELEASE.md:
image

Tag a reviewer: @ayoayco 

Tasks:
- [x] I have ran the build command to make sure apps are working: `npm run build`
- [x] I have ran the tests to make sure nothing is broken: `npm run test`
- [x] I have ran the linter to make sure code is clean: `npm run lint`
- [x] I have reviewed my own code to remove changes that are not needed

### CODE REVIEW

**Review by Icyscools — CHANGES_REQUESTED** [review]

File: packages/pr-updater.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+    if (type == '1') {
+        version = semver.inc(formJSON.version, 'patch');
+    }
+    else if (type == '2') {
+        version = semver.inc(formJSON.version, 'minor');
+    }
+    else {
+        version = semver.inc(formJSON.version, 'major');
+    }
```
[the remark this answers is not in the document]
Comment (ayoayco) [reply packages/pr-updater.js:24]: Yeah, switch-case makes sense here, then maybe a `default` case to throw an error or loop through the question again after a message "Invalid option"

**Review by ayoayco — COMMENTED** [review]
Thanks @NeilAn99, this is great progress for the project. We want to get this in! Can you go through a couple of suggestions first?

**Comment by ayoayco** [comment]
Oh and one very important thing I missed: the package.json "version" property in the package needs to be updated to the same as the version that is written to the RELEASE.md. @NeilAn99

File: packages/pr-updater.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    2. Form 
+    3. Validator`);
+    const noChanged = prompt(": ");
+
+    console.log(`What is the type of PR? 
+    1. Fix
+    2. Minor Release/Feat
+    3. Major Release/Feat`);
+    const type = prompt(": ");
+
```
[the remark this answers is not in the document]
Comment (ayoayco) [reply packages/pr-updater.js:55]: @Fawrish#0487 you can remove adding the version for now in your script. Just focus on the change and type. Where the type could either be "fix" or "feature"

So no need to add the big text for version but just items like:

Features
- Feature description text

Fixes
- Change description text
- dix description 2

**Comment by NeilAn99** [comment]
I have updated the code so that it doesn't deal with versioning, added some input checking and focused on features/fixes. Now running the script will look like:
image
and will produce:
image
Multiple updates to the same package will look like:
image
Is this acceptable? @Icyscools @ayoayco

**Review by Icyscools — APPROVED** [review]
LGTM! 🚀 However, please fix a conflict on a `package.json`. I think it was a line in the `publish` script. Make sure to use it from our latest main branch and it'll be safe to merge. 🦺

**Review by Icyscools — APPROVED** [review]

**Comment by ayoayco** [comment]
Thanks @Icyscools and @NeilAn99 -- I'm merging this now! :)

**Pull request merged**